mesonbuild/backend/ninjabackend.py

```diff
@@ -2307,7 +2307,16 @@ def generate_link(self, target, outfile, outname, obj_list, linker, extra_args=[
         commands += linker.get_option_link_args(self.environment.coredata.compiler_options)
         # Set runtime-paths so we can run executables without needing to set
         # LD_LIBRARY_PATH, etc in the environment. Doesn't work on Windows.
+        if '/' in target.name or '\\' in target.name:
+            # Target names really should not have slashes in them, but
+            # unfortunately we did not check for that and some downstream projects
+            # now have them. Once slashes are forbidden, remove this bit.
+            target_slashname_workaround_dir = os.path.join(os.path.split(target.name)[0],
+                                                       self.get_target_dir(target))
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->